agenda/presentation/src/main/java/com/gkp/agenda/presentation/compoments/EventVisitorsSection.kt

```diff
@@ -0,0 +1,266 @@
+                shape = RoundedCornerShape(5.dp)
+            ),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        Text(
+            text = visitor,
+            style = MaterialTheme.typography.bodyMedium.copy(
+                fontSize = 16.sp
+            )
+        )
+        IconButton(onClick = {}) {
+            Icon(
+                imageVector = Icons.Outlined.Delete,
+                contentDescription = null,
+                tint = TaskyGray
+            )
+        }
+    }
+}
+
+@Composable
+fun AddVisitorDialog(
+    modifier: Modifier = Modifier,
+    onDismiss: () -> Unit,
+    showDialog: Boolean,
+    onAddVisitorClick: (String) -> Unit,
+) {
+    val visitorNameTextState = rememberTextFieldState()
+
+    if (showDialog) {
+        Dialog(
+            onDismissRequest = {
+                onDismiss()
+            }
+        ) {
+            Column(
+                modifier = modifier
+                    .fillMaxWidth()
+                    .padding(16.dp)
+                    .background(
+                        color = MaterialTheme.colorScheme.surface,
+                        shape = MaterialTheme.shapes.extraLarge
+                    )
+            ) {
+                Text(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(16.dp),
+                    text = stringResource(R.string.add_visitor),
+                    style = MaterialTheme.typography.titleLarge.copy(
+                        fontWeight = FontWeight.W600,
+                        fontSize = 20.sp
+                    ),
+                    textAlign = TextAlign.Center
+                )
+                Spacer(modifier = Modifier.size(16.dp))
+                TaskyTextField(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(horizontal = 16.dp),
+                    hintText = stringResource(R.string.visitor_name),
+                    textState = visitorNameTextState
+                )
+                Spacer(modifier = Modifier.size(16.dp))
+                PrimaryButton(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(16.dp),
+                    buttonLabel = stringResource(R.string.add_visitor_button_label),
+                    onClick = {
+                        onAddVisitorClick(visitorNameTextState.text.toString())
+                        onDismiss()
+                    }
+                )
+            }
+        }
+    }
+
+}
+
+@Composable
+fun TaskyTextField(
+    modifier: Modifier = Modifier,
+    hintText: String,
+    textState: TextFieldState,
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+    inputTransformation: InputTransformation? = null,
+) {
+    var hasFocus by remember {
+        mutableStateOf(false)
+    }
```

Comment:
For better control over this component, I'd place the `hasFocus` property as an argument to your text field. This will give you better control in the parent Composable over which text field should be focused when you have multiple of them on a screen using `val (a, b, c) = FocusRequester.createRefs()`